fix: remove duplicates from `activeMocks()` and `pendingMocks()`

mbargiel opened the pull request:
The `interceptorScopes()` function is used by `activeMocks()` and `pendingMocks()` to retrieve the list of all scopes. That list is built by concatenating the scopes of every interceptor. When multiple interceptors are registered on the same scope, the scope will be returned multiple times, causing `activeMocks()` and `pendingMocks()` to include duplicates.

This PR fixes the issue by ensuring there is no duplicate returned by `interceptorScopes()`.

Fixes #2352 (and #600, which also reported this issue)

mikicho reviewed (approved):
LGTM. Thanks!

mikicho commented:
I was unable to merge the master to this PR to resolve the conflict, so I will wait for the contributor to do it. If not, I will open a new PR.

mbargiel commented:
Woah, that's quite the necro 😅  I'll see if I can find some time to remember what this was about and solve the merge conflict. (Note: I have two other PRs fixing other bugs that also have merge conflicts.)

mbargiel commented:
Well, it turns out just rebasing worked without any conflict, so here you go, updated PR :) I also locally verified that the new test I added fails on its own (ie running tests at c366ed115e322b14dd885523c41b701cd9e712ac results in my new test failing, then running tests on 0b6ab12568fc8b6c5845bcad02102552bb113815 (branch head) passes).

gr2m commented:
Fixed the formatting and also confirmed that test is indeed failing on [c366ed1](https://github.com/nock/nock/pull/2356/commits/c366ed115e322b14dd885523c41b701cd9e712ac)

gr2m reviewed (approved):
Thank you 💐
